### PR TITLE
Resolve small inconsistencies at Results API

### LINKED ISSUE #5872: Small inconsistencies at Java Results API
### Play Version

2.5.0

### API

Java

### Actual Behavior

1. [`StatusHeader.sendFile`](https://github.com/playframework/playframework/blob/58a150d/framework/src/play/src/main/java/play/mvc/StatusHeader.java#L170) and [`StatusHeader.sendResource`](https://github.com/playframework/playframework/blob/58a150d/framework/src/play/src/main/java/play/mvc/StatusHeader.java#L94) both have `inline=true` as default and [`StatusHeader.sendPath`](https://github.com/playframework/playframework/blob/58a150d/framework/src/play/src/main/java/play/mvc/StatusHeader.java#L130) has **`inline=false`** as default.
2. There is a [`sendPath(Path path, boolean inline, String filename)`](https://github.com/playframework/playframework/blob/58a150d/framework/src/play/src/main/java/play/mvc/StatusHeader.java#L152), but there is no equivalent `sendFile` or `sendResource`. Also, Scala Results API have methods to set both if the File/Path will be delivered inline and the content disposition filename.
3. [`sendInputStream(InputStream stream)`](https://github.com/playframework/playframework/blob/58a150d/framework/src/play/src/main/java/play/mvc/StatusHeader.java#L55) uses a `HttpEntity.chunked` and [`sendInputStream(InputStream stream, long contentLength)`](https://github.com/playframework/playframework/blob/58a150d/framework/src/play/src/main/java/play/mvc/StatusHeader.java#L70) uses a `HttpEntity.Streamed` (`File` and `Path` methods [uses `HttpEntity.Streamed`](https://github.com/playframework/playframework/blob/58a150d/framework/src/play/src/main/java/play/mvc/StatusHeader.java#L209)).

### PR DESCRIPTION
**Pull request by marcospereira**
## Pull Request Checklist

* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you added tests for any changed functionality?

## Fixes

Fixes #5872.

## Purpose

This made Java Results API more consistent while sending files/paths and resources. It was also checked against Scala API to ensure consistent between the two APIs (at least in relation to sending files, paths and resources).

## References

See #5872.

### CODE REVIEW

File: framework/src/play/src/main/java/play/mvc/StatusHeader.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 public class StatusHeader extends Result {
 
     private static final int defaultChunkSize = 1024 * 8;
+    private static final boolean defaultInlineMode = false;
```
Comment (benmccann) [inline framework/src/play/src/main/java/play/mvc/StatusHeader.java]: constants in Java would typically be written as `DEFAULT_INLINE_MODE`

Comment (marcospereira) [inline framework/src/play/src/main/java/play/mvc/StatusHeader.java]: Fixed.

File: framework/src/play/src/main/java/play/mvc/StatusHeader.java [hunk 2]
```diff
@@ -79,7 +80,7 @@ public Result sendInputStream(InputStream stream, long contentLength) {
      * @return an inlined '200 OK' result containing the resource in the body with in-line content disposition.
```
Comment (benmccann) [inline framework/src/play/src/main/java/play/mvc/StatusHeader.java]: the method comments are out-of-date

File: framework/src/play/src/main/java/play/mvc/StatusHeader.java [hunk 3]
```diff
@@ -79,7 +80,7 @@ public Result sendInputStream(InputStream stream, long contentLength) {
...
      */
     public Result sendResource(String resourceName) {
-        return sendResource(resourceName, true);
+        return sendResource(resourceName, DEFAULT_INLINE_MODE);
```
Comment (benmccann) [inline framework/src/play/src/main/java/play/mvc/StatusHeader.java]: should add something to the migration guide to let people know the behavior of this method is changing?

**Comment by marcospereira** [comment]
@benmccann I think this is good to be merged. :-)

**Comment by benmccann** [comment]
The only issue I have with this is setting `DEFAULT_INLINE_MODE` to `false`. I think that inline is a better user experience and thus probably a better default. I also think that changing the behavior if existing methods isn't great for backwards compatibility. We should at least document it in the migration guide. But if you make `true` the default then I think you get the better experience and avoid the backwards compatibility issue.

Everything else looks good.

**Comment by marcospereira** [comment]
@benmccann,

Here is the current state of things (not considering this PR):

| API   | Method             | Default        |
|:------|:-------------------|:---------------|
| Scala | `sendResource`    | `inline=true`  |
| Scala | `sendPath`        | `inline=false` |
| Scala | `sendFile`        | `inline=false` |
| Java  | `sendInputStream` | `none`         |
| Java  | `sendResource`    | `inline=true`  |
| Java  | `sendPath`        | `inline=false` |
| Java  | `sendFile`        | `inline=true`  |

This looks accidental to me, so I want to change it and have a default that is shared between all methods, including Scala and Java APIs. Of course, it has to be a sensible default and, based on your explanation, `inline=true` sounds like a good default to me.

WDYT?

**Comment by mkurz** [comment]
I share the concerns @benmccann has. The inline mode changes should be documenened in the migration guide.
Do you think it would make sense to let the user set a default for the inline mode in `application.conf`?

**Comment by marcospereira** [comment]
@mkurz and @benmccann now both Scala and Java API are using `inline` as the default behavior. This was also documented at migration guide.

@mkurz I don't see a strong reason to make this configurable (since there was not a consistent default to which people could migrate), so I did not changed that.

File: framework/src/play/src/main/java/play/mvc/StatusHeader.java [hunk 4]
```diff
@@ -200,14 +240,26 @@ public Result sendFile(File file, boolean inline) {
      * @return a '200 OK' result containing the file
      */
     public Result sendFile(File file, String fileName) {
+        return sendFile(file, DEFAULT_INLINE_MODE, fileName);
+    }
+
+    /**
+     * Send the given file as an attachment.
```
Comment (benmccann) [inline framework/src/play/src/main/java/play/mvc/StatusHeader.java]: Remove "as an attachment". It may or may not be

**Comment by benmccann** [comment]
LGTM

File: framework/src/play/src/main/java/play/mvc/StatusHeader.java (hunk 4 above)
Comment (marcospereira) [inline framework/src/play/src/main/java/play/mvc/StatusHeader.java]: Fixed. :-)

**Comment by benmccann** [comment]
Build is failing. You need to update `play.mvc.ResultsTest`

**Comment by marcospereira** [comment]
@benmccann build is green again.

**Pull request merged**